diff --git a/src/Plugin/views/style/DataExport.php b/src/Plugin/views/style/DataExport.php
index 100bc59..3b53b4b 100644
--- a/src/Plugin/views/style/DataExport.php
+++ b/src/Plugin/views/style/DataExport.php
@@ -239,17 +239,17 @@ class DataExport extends Serializer {
 
     $url = $url->setOptions($url_options)->toString();
 
-    // Add the CSV icon to the view.
-    $type = $this->displayHandler->getContentType();
+    // Add the icon to the view.
+    $format = $this->displayHandler->getContentType();
     $this->view->feedIcons[] = [
       '#theme' => 'export_icon',
       '#url' => $url,
-      '#type' => mb_strtoupper($type),
+      '#format' => mb_strtoupper($format),
       '#theme_wrappers' => [
         'container' => [
           '#attributes' => [
             'class' => [
-              Html::cleanCssIdentifier($type) . '-feed',
+              Html::cleanCssIdentifier($format) . '-feed',
               'views-data-export-feed',
             ],
           ],
diff --git a/templates/export-icon.html.twig b/templates/export-icon.html.twig
index dba6b4d..4126211 100644
--- a/templates/export-icon.html.twig
+++ b/templates/export-icon.html.twig
@@ -6,12 +6,12 @@
  * Available variables:
  * - url: An internal system path of the data export.
  * - attributes: Remaining HTML attributes for the export link.
- *   - type: The type of export (csv, json, xml).
+ *   - format: The export format (csv, json, xml).
  *   - class: HTML classes to be applied to the export link.
  *
  * @ingroup themeable
  */
 #}
 <a href="{{ url }}"{{ attributes.addClass('feed-icon') }}>
-  {{ 'Download @type'|t({'@type': type}) }}
+  {{ 'Download @format'|t({'@format': format}) }}
 </a>
diff --git a/views_data_export.module b/views_data_export.module
index e910819..bc3fa46 100644
--- a/views_data_export.module
+++ b/views_data_export.module
@@ -12,7 +12,7 @@ function views_data_export_theme() {
   return [
     'export_icon' => [
       'variables' => [
-        'type' => NULL,
+        'format' => NULL,
         'url' => NULL,
       ],
     ],
