From fdeea219ebfcef20ad62656c1dba4e9d74977a3c Mon Sep 17 00:00:00 2001
From: MatthieuScarset <m@matthieuscarset.com>
Date: Tue, 22 May 2018 15:34:27 -0400
Subject: [PATCH 1/1] Load entity in term_condition plugin form

---
 src/Plugin/Condition/Term.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Plugin/Condition/Term.php b/src/Plugin/Condition/Term.php
index 1935a69..7d901cb 100644
--- a/src/Plugin/Condition/Term.php
+++ b/src/Plugin/Condition/Term.php
@@ -10,6 +10,7 @@ namespace Drupal\term_condition\Plugin\Condition;
 use Drupal\Core\Condition\ConditionPluginBase;
 use Drupal\Core\Form\FormStateInterface;
 use Drupal\Core\Plugin\Context\ContextDefinition;
+use Drupal\taxonomy\Entity\Term as CoreTerm;
 
 /**
 * Provides a 'Term' condition to enable a condition based in module selected status.
@@ -60,7 +61,7 @@ class Term extends ConditionPluginBase {
     $form['tid'] = array(
       '#type' => 'entity_autocomplete',
       '#title' => $this->t('Select a taxonomy term'),
-      '#default_value' => $this->configuration['tid'],
+      '#default_value' => isset($this->configuration['tid']) ? CoreTerm::load($this->configuration['tid']) : NULL,
       '#target_type' => 'taxonomy_term',
     );
 
-- 
2.17.0

